components/TextFields/src/ContainedInputView/MDCBaseTextField.h

```diff
@@ -29,6 +29,11 @@
  */
 @property(strong, nonatomic, readonly, nonnull) UILabel *label;
 
+/**
+ The color of the placeholder. The default is @c .lightGrayColor .
+ */
+@property(nonatomic, strong, nonnull) UIColor *placeholderColor;
```

Comment:
Putting down thoughts discussed in person:
As written this is a magic property that changes another property: attributedPlaceholder. Properties should not change other properties. 

Instead we should consider implementing this as a category on the UITextField as a convenience method following the mutator pattern.
If we are use the public API of the UIKit object let us try to accomplish it as a category on the UIKit object.

Reply:
@randallli, the current solution does rely on side effects, but it does so in a way that I think is unlikely to get in the way of users' intentions. With the current solution, setting `placeholder` or `placeholderColor` will result in a call to `-setAttributedPlaceholder:`, but it doesn't affect clients' ability to make use of that API on its own. It would behave like this:

```objc
NSDictionary *redAttribute = @{NSForegroundColorAttributeName: [UIColor redColor]};
NSAttributedString *attributedString = [[NSAttributedString alloc] initWithString:@"C" attributes:redAttribute];

self.textField.placeholder = @"A";
self.textField.placeholderColor = [UIColor blueColor];
// placeholder will be a blue "A"
self.textField.placeholder = @"B";
// placeholder will be a blue "B"
self.textField.attributedPlaceholder = attributedString;
// placeholder will be a red "C"
self.textField.placeholderColor = [UIColor blueColor];
// placeholder will be a blue "C"
```
The solution we talked about yesterday involved a UITextField category class method like `+setPlaceholderColor:onTextField:`. That will behave like this: 

```objc
self.textField.placeholder = @"A";
[UITextField setPlaceholderColor:[UIColor blueColor] onTextField:self.textField];
// placeholder will be a blue "A"
self.textField.placeholder = @"B";
// placeholder will be a gray "B"
```

Do you definitely think the second one is better?

Reply:
After looking at your examples I think a better signature would be a convenience setter that documents that it is changing the placeholderAttributed property.
it would look something like this:
[self.textField setPlaceholder:@"A" withColor:[UIColor redColor]]

it would be a UITextField category without a getter.

what do you think?

Reply:
I downscoped the PR to get it landed. There's no longer a placeholderColor property.
Also, something I said about UITextField behavior in the last comment was inaccurate.

This is how UITextField really behaves:

```
textField.attributedPlaceholder = [self attributedStringWithString:@"C" color:[UIColor redColor]];
// placeholder will be a red "C"
textField.placeholder = @"Q";
// placeholder will be a red "Q"
textField.attributedPlaceholder = nil;
// placeholder doesn't exist
textField.placeholder = @"B";
// placeholder will be a red "B"
textField.attributedPlaceholder = [self attributedStringWithString:@"Z" color:[UIColor greenColor]];
// placeholder will be a green "Z"
```